packages/eas-cli/src/metadata/apple/config/reader.ts

```diff
@@ -78,14 +78,33 @@ export class AppleConfigReader {
   }
 
   public getCategories(): CategoryIds | null {
-    if (Array.isArray(this.schema.categories) && this.schema.categories.length > 0) {
-      return {
-        primaryCategory: this.schema.categories[0],
-        secondaryCategory: this.schema.categories[1],
-      };
+    const { categories } = this.schema;
+    if (!categories || categories.length <= 0) {
+      return null;
     }
 
-    return null;
+    // We validate the categories based on enums, but they will still be strings here.
+    const categoryIds: Partial<Record<keyof CategoryIds, string>> = {};
+
+    if (Array.isArray(categories[0])) {
+      categoryIds.primaryCategory = categories[0][0];
+      categoryIds.primarySubcategoryOne = categories[0][1];
+      categoryIds.primarySubcategoryTwo = categories[0][2];
+    } else {
+      categoryIds.primaryCategory = categories[0];
+    }
+
+    if (Array.isArray(categories[1])) {
+      categoryIds.secondaryCategory = categories[1][0];
+      categoryIds.secondarySubcategoryOne = categories[1][1];
+      categoryIds.secondarySubcategoryTwo = categories[1][2];
+    } else {
+      categoryIds.secondaryCategory = categories[1];
+    }
+
+    // Because we handle categories as normal strings,
+    // the type doesn't match with the actual CategoryIds types.
+    return categoryIds as CategoryIds;
```

Comment:
I don't get it, if categories are strings why would we want to pretend they are objects? Maybe we should alter the type?